subworkflows/local/templates.nf

```diff
@@ -0,0 +1,49 @@
+include { CREATE_TCOFFEETEMPLATE } from '../../modules/local/create_tcoffee_template'
+
+workflow TEMPLATES {
+
+    take:
+    ch_dependencies // channel: [ meta, /path/to/file.* ]
+    ch_templates // channel: [ meta, /path/to/template.txt ]
+    suffix
+
+    main:
+
+    ch_versions     = Channel.empty()
+
+    ch_dependencies_template = ch_dependencies.join(ch_templates, by:0, remainder:true)
+    ch_dependencies_template
+        .branch {
+            template: it[2] != null
+            no_template: true
+        }
+        .set { ch_dependencies_branched }
+
+    // Create the new templates and merge them with the existing templates
+    CREATE_TCOFFEETEMPLATE (
```

Comment:
The linter will probably complain that there is only one process here, but it is better to have this code separated for scalability

Reply:
Yes I saw it, it does indeed complain but I discussed with Cedric and he has strong feelings for this design and now I do agree with him too.